command/jsonprovider/block.go

```diff
@@ -0,0 +1,69 @@
+package jsonprovider
+
+import (
+	"github.com/hashicorp/terraform/configs/configschema"
+)
+
+type block struct {
+	Attributes map[string]*attribute `json:"attributes,omitempty"`
+	BlockTypes map[string]*blockType `json:"block_types,omitempty"`
+}
+
+type blockType struct {
+	NestingMode string `json:"nesting_mode,omitempty"`
+	Block       *block `json:"block,omitempty"`
+	MinItems    uint64 `json:"min_items"`
+	MaxItems    uint64 `json:"max_items"`
+}
+
+func marshalBlockTypes(nestedBlock *configschema.NestedBlock) *blockType {
+	if nestedBlock == nil {
+		return &blockType{}
+	}
+	ret := &blockType{
+		Block:    marshalBlock(&nestedBlock.Block),
+		MinItems: uint64(nestedBlock.MinItems),
+		MaxItems: uint64(nestedBlock.MaxItems),
+	}
+
+	switch nestedBlock.Nesting.String() {
```

Comment:
I was surprised to see the `.String()` call here vs. switching directly on the `nestedBlock.Nesting` value. Could you add a comment explaining it?

Reply:
I'll just drop the `.String()` and check the nested mode directly - I had doubts when I wrote that but forgot to follow up.